### DISCUSSION TITLE
Secondary color cannot be used as an accent color like in Material Design 2 with the new theming system

### DISCUSSION THREAD

**Discussion by TaneliTuomola** [primefaces/community#2886] [PrimeVue] [answered]
In the Material Design 2 specification, the secondary color is used as an accent color with a heavy emphasis, but with the new Prime theming system, secondary button must be gray, because the secondary button has been used inside other components such as the dialog close button or pick list arrows.

I suggest that it be possible to change the color of buttons inside other components, such as the dialog close button, or to use the "plain" variant instead of the "secondary" button. This would allow prime themes secondary color be used more freely. @cagataycivici 

Screenshot of the old style material design theme on primeNg:
<img width="329" alt="image" src="https://github.com/user-attachments/assets/69433a2e-910b-41f2-accd-7287c4d4be7e">

**Comment by cagataycivici** [comment]
Primary colors are dynamic now so we could not settle on a specific secondary color by default, however you can easily customize it with the **button.secondary.*** design tokens.

The button docs, lists the tokens for secondary.

https://primevue.org/button/

Also please see;

https://primevue.org/theming/styled/#component

```js
const MyPreset = definePreset(Material, {
    components: {
        button: {
           secondary: {
                background: '{pink.400}'
               ...
           }
        }
    }
});


```

**Reply by TaneliTuomola** [reply]
The problem is when the secondary button has been used inside other components, such as the dialog close button, and these do not yet have their own variables in the presets.

I made a separate issue for this on the primeNg side, for which you are making a material design preset next: https://github.com/primefaces/primeng/issues/16585

**Comment by cagataycivici** [comment, marked as answer]
In that case, you may either create a new **accent** button and not use the secondary. An example doc from Vue, the NG doc is not updated yet;

https://primevue.org/theming/styled/#extend

Sounds like nice tutorial idea too.

**Reply by TaneliTuomola** [reply]
I can't wait to get something similar for primeNg. Of course, in design system use, separate design tokens for all component styles is desirable and makes prime libraries even more design agnostic. Thanks for your replies and keep up the good work!
